Kernel/Term.cpp

```diff
@@ -1381,6 +1403,15 @@ bool Term::isBoolean() const {
           break;
         }
       }
+      case SF_MATCH: {
+        const TermList *ts = term->nthArgument(2);
```

Comment:
Couldn't this crash for a match over the empty type?

Reply:
As per the 2.6 standard and [this](https://github.com/vprover/vampire/blob/mhajdu-smtlib2-new-keywords/Parse/SMTLIB2.cpp#L1018) and [this](https://github.com/vprover/vampire/blob/mhajdu-smtlib2-new-keywords/Parse/SMTLIB2.cpp#L1108) line, declaring an empty type shouldn't be possible and we also add missing ctors when some are covered with a variable pattern.

Reply:
Thank you for the clarification!